src/Features/LanguageServer/Microsoft.CodeAnalysis.LanguageServer/HostWorkspace/LanguageServerProjectSystem.cs

```diff
@@ -148,11 +149,12 @@ private async ValueTask LoadOrReloadProjectsAsync(ImmutableSegmentedList<Project
         {
             var tasks = new List<Task>();
 
+            using var _ = PooledHashSet<string>.GetInstance(out var projectsWithUnresolvedDependencies);
```

Comment:
This needs to be a concurrent set since we're going to be modifying from multiple threads.